File: 2020/01/ts/part-two.ts
import fs from 'fs-extra';

const rawInput = fs.readFileSync('../input.txt')
  .toString();
const input = rawInput.split(`
`)
  .map(Number);

for (let i = 0; i < input.length; i++) {
  for (let j = i + 1; j < input.length; j++) {
    for (let k = j + 1; k < input.length; k++) {
      const num1 = input[i];
      const num2 = input[j];
      const num3 = input[k];
      if (num1 + num2 + num3 === 2020) {
        console.log(num1 * num2 * num3);
        break;
      }

    }

  }
}

File: 2020/01/ruby/part-two.rb
file = File.open("./../input.txt")
input = file.read.split.map{|item| item.to_i}
file.close


for i in 0..input.length do
  for j in (i+1)..input.length do
    for k in (j+1)..input.length do
      num1 = input[i]
      num2 = input[j]
      num3 = input[k]

      if num2.nil? || num3.nil? then
        next
      end

      sum = num1 + num2 + num3

      if(sum==2020)
        puts num1 * num2 * num3
      end
    end
  end
end